src/commands/Minion/trekshop.ts

```diff
@@ -0,0 +1,172 @@
+import { CommandStore, KlasaMessage } from 'klasa';
+import { Bank } from 'oldschooljs';
+import { bankHasAllItemsFromBank } from 'oldschooljs/dist/util';
+
+import { Time } from '../../lib/constants';
+import TrekShopItems, { TrekExperience } from '../../lib/data/buyables/trekBuyables';
+import { rewardTokens } from '../../lib/minions/data/templeTrekking';
+import { AddXpParams } from '../../lib/minions/types';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import { SkillsEnum } from '../../lib/skilling/types';
+import { BotCommand } from '../../lib/structures/BotCommand';
+import { percentChance, rand, reduceNumByPercent, stringMatches, toTitleCase } from '../../lib/util';
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			usage: '<easy|medium|hard> [quantity:integer{1,2147483647}] <item:...string>',
+			usageDelim: ' ',
+			oneAtTime: true,
+			cooldown: 5,
+			altProtection: true,
+			aliases: ['ts'],
+			categoryFlags: ['minion'],
+			description: 'Allows you to redeem reward tokens from Temple Trekking.',
+			examples: ['+ts easy bowstring', '+ts 5  hard herbs']
+		});
+	}
+
+	async run(msg: KlasaMessage, [type = 'easy', quantity, name]: ['easy' | 'medium' | 'hard', number, string]) {
+		const user = msg.author;
+		await user.settings.sync(true);
+
+		const userBank = user.settings.get(UserSettings.Bank);
```

Comment:
you should use the Bank class instead, and make appropriate changes below, e.g. `userBank.amount()` instead of `userBank[itemID]` and `userBank.has(something.bank)` instead of `bankHasAllItemsFromBank`.